### PR TITLE
Ensure that comments are stripped before reading tasks.json file

### LINKED ISSUE #1074: 1.6.0: Extension host terminated unexpectedly. Please reload the window to recover.
## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.0-preview2-003121)

Product Information:
 Version:            1.0.0-preview2-003121
 Commit SHA-1 hash:  1e9d529bc5

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64

VS Code version:
1.8.1
C# Extension version:
1.6.0
## Steps to reproduce
Open a C# file. Omnisharp is starting up and when it's almost finished: "Extension host terminated unexpectedly. Please reload the window to recover."

## Expected  behavior
No crash and a working C# environment. I don't see an exception in the Omnisharp log. It ends with:

```
[INFORMATION:OmniSharp#MSBuild] MSBUILD_EXE_PATH environment variable set to /Users/tverboon/.vscode/extensions/ms-vscode.csharp-1.6.0/bin/omnisharp/msbuild/MSBuild.dll
[INFORMATION:OmniSharp#MSBuild] MSBuildExtensionsPath environment variable set to /Users/tverboon/.vscode/extensions/ms-vscode.csharp-1.6.0/bin/omnisharp/msbuild
[INFORMATION:OmniSharp#MSBuild] MSBuildSDKsPath environment variable set to /Users/tverboon/.vscode/extensions/ms-vscode.csharp-1.6.0/bin/omnisharp/msbuild/Sdks
[INFORMATION:OmniSharp#MSBuild] No solution files found in '/Users/tverboon/Desktop/Code/Midmid'
[INFORMATION:OmniSharp.Script.ScriptProjectSystem] Detecting CSX files in '/Users/tverboon/Desktop/Code/Midmid'.
[INFORMATION:OmniSharp.Script.ScriptProjectSystem] Could not find any CSX files
[INFORMATION:OmniSharp.Startup] Configuration finished.
```

## Actual behavior
"Extension host terminated unexpectedly. Please reload the window to recover." And C# is not working.

---
Extra info:
I did some extra testing as I had to reinstall the previous version. For me the extension causes a crash in 1.6-beta7. 1.6-beta6 and before are working.

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #1074

### CODE REVIEW

File: src/assets.ts [hunk 1]
```diff
@@ -357,7 +358,7 @@ function getBuildOperations(tasksJsonPath: string) {
                     }
 
                     const text = buffer.toString();
-                    const tasksJson: tasks.TaskConfiguration = JSON.parse(text);
+                    const tasksJson: tasks.TaskConfiguration = tolerantParse(text);
```
Comment (gregg-miskelly) [inline src/assets.ts]: Should this have a try/catch around it?

File: src/json.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            parts.push(text.substring(partStart, length));
+            break;
+        }
+    }
+
+    return parts.join('');
+}
+
+export function tolerantParse(text: string) {
+    text = stripComments(text);
```
Comment (gregg-miskelly) [inline src/json.ts]: Should we ignore the UTF8-bom while we are at it too?

Comment (DustinCampbell) [inline src/json.ts]: Yes, we should. Also, I'd like to add tolerance around trailing commas as well.

File: src/assets.ts (hunk 1 above)
Comment (DustinCampbell) [inline src/assets.ts]: It should, but it didn't before. I'll add one.

File: src/json.ts (hunk 2 above)
Comment (DustinCampbell) [inline src/json.ts]: OK. Added code to be tolerant against both byte order marks and trailing comments in object member or array element lists.

File: src/json.ts [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                }
+
+                break;
+
+            case CharCode.comma:
+                // Ignore trailing commas in object member lists and array element lists
+                let nextCode = peekPastWhitespace();
+                if (nextCode === CharCode.closeBrace || nextCode === CharCode.closeBracket) {
+                    parts.push(text.substring(partStart, index - 1));
+                    partStart = index;
```
Comment (gregg-miskelly) [inline src/json.ts]: don't you need to know the index of the comma in the case that peekPastWhitespace did something?

Comment (DustinCampbell) [inline src/json.ts]: `peekPastWhitespace` doesn't do anything but look ahead. It doesn't modify `index`

File: test/json.test.ts [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        "element1",
+        "element2"
+    ]
+}`;
+
+        let json = tolerantParse(text);
+        let result = JSON.stringify(json, null, 4);
+
+        result.should.equal(expected);
+    });
```
Comment (gregg-miskelly) [inline test/json.test.ts]: Add a test for something where the extra `,` is preceeded by whitespace? (ex: 

```json
{
   "obj" : { "a": 1 , }
}
```

Comment (DustinCampbell) [inline test/json.test.ts]: Good idea. Done.

File: src/json.ts (hunk 3 above)
Comment (gregg-miskelly) [inline src/json.ts]: Oh, never mind. I was thinking that partStart was going to get the wrong value, but this works.

**Comment by gregg-miskelly** [comment]
LGTM

File: test/json.test.ts (hunk 4 above)
Comment (DustinCampbell) [inline test/json.test.ts]: Just thought of an issue. This won't work when a trailing comma has a comment at the end of the line. Fixing...

Comment (DustinCampbell) [inline test/json.test.ts]: Actually, I'm going to let that go for now.

Comment (DustinCampbell) [inline test/json.test.ts]: It'll require significant refactoring

Comment (DustinCampbell) [inline test/json.test.ts]: Filed #1076 to track this.
